### PR TITLE
Remove goog.string

### PR DESCRIPTION
**Pull request by nicholas-l**
Added a utils.js file which has one function in it at the moment (padNumber) or would this be better suited in a string.js file?

### CODE REVIEW

**Comment by nicholas-l** [comment]
Also not sure how to tackle compareVersions, could we have a very simplify case and only deal with numeric types? i.e. not have to deal with "1.2b"

**Comment by ahocevar** [comment]
Looks good. +1 on only handling numeric versions. Just one thing: could you rename utils to string? Also the file name.

**Comment by nicholas-l** [comment]
Done :) not sure about the numeric versions. I put tests in for it and they seem to pass :) So ready for review

File: test/spec/ol/utils.test.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+goog.provide('ol.test.utils');
```
Comment (tschaub) [inline test/spec/ol/utils.test.js]: This should be `ol.test.string` and the file should be named `string.test.js`.

File: test/spec/ol/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      expect(ol.string.padNumber(9.9, 2, 0)).to.be('10');
+      expect(ol.string.padNumber(5, 0, 0)).to.be('5');
+      expect(ol.string.padNumber(5, 1, 1)).to.be('5.0');
+      expect(ol.string.padNumber(5, 2, 1)).to.be('05.0');
+      expect(ol.string.padNumber(5, 2, 0)).to.be('05');
+    });
+
+  });
+
+  describe('ol.string.padNumber', function() {
```
Comment (tschaub) [inline test/spec/ol/utils.test.js]: This should be `ol.string.compareVersions()`.

File: test/spec/ol/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    var f = ol.string.compareVersions;
+    expect(f(1, 1)).to.be(0);
```
Comment (tschaub) [inline test/spec/ol/utils.test.js]: These assertions need to be in an `it()` call for failures to be properly caught and reported.

File: test/spec/ol/utils.test.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    expect(f('1', '1.1')).to.be.below(0);
+    expect(f('2.2', '2')).to.be.above(0);
+
+    // expect(f('1.1', '1.1b')).to.be.above(0);
+    // expect(f('1.1b', '1.1')).to.be.below(0);
+    // expect(f('1.1b', '1.1b')).to.be(0);
+
+    // expect(f('1.1b', '1.1a')).to.be.above(0);
+    // expect(f('1.1a', '1.1b')).to.be.below(0);
```
Comment (tschaub) [inline test/spec/ol/utils.test.js]: Please remove the commented out code.

**Comment by tschaub** [comment]
This looks good @nicholas-l.  Thanks for the work on it.  A few comments above about the tests.  Any failures in `ol.string.compareVersions()` assertions aren't currently being caught.

**Comment by nicholas-l** [comment]
@tschaub changed according to comments

**Comment by fredj** [comment]
@nicholas-l can you please squash all your commits into one? (there a good explanation on how to do that here: https://github.com/openlayers/ol3/pull/4868#issuecomment-187785482)

**Comment by nicholas-l** [comment]
@fredj squashed

**Comment by fredj** [comment]
Thanks @nicholas-l

**Pull request merged**